src/hooks/useSnapshotDelegateContract.ts

```diff
@@ -0,0 +1,38 @@
+import { hexlify } from '@ethersproject/bytes'
+import { Contract } from '@ethersproject/contracts'
+import { Web3Provider } from '@ethersproject/providers'
+import useAuth from 'decentraland-gatsby/dist/hooks/useAuth'
+import { useMemo } from 'react'
+import { SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
+import DelegateABI from '../modules/contracts/abi/Delegate.abi.json'
+
+function useSnapshotDelegateContract() {
+
+  const [userAddress, authState] = useAuth()
+
+  return useMemo(() => {
```

Comment:
Is this memo necessary? I think we can safely remove it.

Reply:
When `authState.provider` changes we should update all methods right?

Reply:
I used the memo thinking about what Lemu says, I do not see how it could be safely removed